applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -189,94 +189,86 @@ void SmallStrainUPwDiffOrderElement::Initialize(const ProcessInfo& rCurrentProce
 {
     KRATOS_TRY
 
-    const GeometryType&                             rGeom = GetGeometry();
-    const GeometryType::IntegrationPointsArrayType& IntegrationPoints =
-        rGeom.IntegrationPoints(this->GetIntegrationMethod());
-
-    if (mConstitutiveLawVector.size() != IntegrationPoints.size())
-        mConstitutiveLawVector.resize(IntegrationPoints.size());
+    const auto& r_properties = GetProperties();
+    const auto& r_geometry   = GetGeometry();
+    const auto number_of_integration_points = r_geometry.IntegrationPointsNumber(GetIntegrationMethod());
 
-    if (GetProperties()[CONSTITUTIVE_LAW] != nullptr) {
```

Comment:
This check is now removed, so in principle we could get a seg fault here. Should we still add a KRATOS_ERROR_IF(GetProperties()[CONSTITUTIVE_LAW] != nullptr)?

Reply:
When I compared this implementation with the non-diff order counterpart, I noticed that this check was missing there. Since this property is supplied by the user, it should have been checked way earlier. Therefore, I have decided to remove this check, to avoid that we keep checking input over and over again. If it segfaults at this point, we are missing a crucial check at the input validation stage.